packages/backend/server/src/modules/index.ts

```diff
@@ -22,30 +22,40 @@ switch (SERVER_FLAVOR) {
   case 'sync':
     BusinessModules.push(SyncModule, DocModule.forSync());
     break;
+  case 'selfhosted':
```

Comment:
Actually I have practiced before about module plugin/out system.

Just like how Gitlab did, we could make it an Application Level configs stored in DB, and before client init modules, it should fetching the features list from server so it can know which modules should be enabled and whichs are not.

Reply:
make backend module load optional and configable is not complicated, but currently, many requests on the frontend are executed in the background and cannot be skipped or fault-tolerant, if backend disable some module, frontend will fault when request the apis of the corresponding module. i think we need reduce frontend data stream first
in currently stage, only payment module need to disable in come scene(mainly in selfhosted), i think provide a new flavor of server is acceptable

Reply:
make a plan

Reply:
https://github.com/toeverything/AFFiNE/issues/5057 already make a issue